app/javascript/react/components/molecules/Calendar/Calendar.tsx

```diff
@@ -1,27 +1,190 @@
+import moment, { Moment } from "moment";
 
-import { selectLatestThreeMonthsDailyAverages } from "../../../store/fixedStreamSelectors";
+import { selectThreeMonthsDailyAverage } from "../../../store/movingStreamSelectors";
 import { Month } from "./atoms/Month";
-import * as S from "./Calendar.style";
 import { useTranslation } from "react-i18next";
 import HeaderToggle from "./HeaderToggle/HeaderToggle";
+import { useAppDispatch } from "../../../store/hooks";
+import {
+  movingData,
+  fetchNewMovingStream,
+} from "../../../store/movingCalendarStreamSlice";
+import { ScrollCalendarButton } from "./atoms/ScrollCalendarButton/ScrollCalendarButton";
+import { MovesKeys } from "../../../types/movesKeys";
+import * as S from "./Calendar.style";
+
+interface MovableCalendarData {
+  zeroDate: string;
+  currentStartDate: string;
+  currentEndDate: string;
+  direction: number | undefined;
+  triggerDirectionUpdate: number;
+}
 
-const Calendar = () => {
-  const threeMonthsData = useSelector(selectLatestThreeMonthsDailyAverages);
+interface CalendarProps {
+  streamId: number;
+}
+
+const Calendar: React.FC<CalendarProps> = ({ streamId }) => {
+  const SEEN_MONTHS_NUMBER = 3;
+
+  const threeMonthsData = useSelector(selectThreeMonthsDailyAverage);
+  const movingCalendarData = useSelector(movingData);
   const { t } = useTranslation();
+  const dispatch = useAppDispatch();
+  const [isButtonDisabled, setIsButtonDisabled] = useState(false);
+  const [dateReference, setDateReference] = useState<MovableCalendarData>({
+    zeroDate: "",
+    currentStartDate: "",
+    currentEndDate: "",
+    direction: undefined,
+    triggerDirectionUpdate: 0,
+  });
+
+  const updateMoveValue = (direction: MovesKeys) => {
+    setDateReference((prevState) => {
+      return {
+        ...prevState,
+        direction: direction,
+        triggerDirectionUpdate: prevState.triggerDirectionUpdate + 1,
+      };
+    });
+  };
+
+  const handleBackwardMovement = (dateMoment: moment.Moment) => {
+    const newEndMoment = dateMoment.subtract(1, "months").endOf("month");
+    setIsButtonDisabled(false);
+    return newEndMoment;
+  };
+
+  const handleForwardMovement = (dateMoment: moment.Moment) => {
+    let newEndMoment = dateMoment.add(1, "months").endOf("month");
+    const zeroDateMoment = moment(dateReference.zeroDate, "YYYY-MM-DD");
+    if (newEndMoment.isAfter(zeroDateMoment)) {
+      newEndMoment = zeroDateMoment;
+      setIsButtonDisabled(true);
+    }
+    return newEndMoment;
+  };
+
+  useEffect(() => {
+    // Disable forward button at first, becouse
+    // we present the newest months just after loading this page
+    setIsButtonDisabled(true);
+
+    const lastElementIdx = movingCalendarData.data.length - 1;
+    const endDate = movingCalendarData.data[lastElementIdx].date;
+
+    const startMoment = moment(endDate, "YYYY-MM-DD");
+    const newStartDate = startMoment
+      .date(1)
+      .subtract(SEEN_MONTHS_NUMBER - 1, "months")
```

Comment:
Is it possible to move 1 and -1 to constants to make the code more understandable?

Reply:
Honestly, I do not know how to even call this const. We need to subtract 1, because we want to see 3 months, but we start from some ex. X so we only want to get back in 2 months, not more that's why